scripts/lint_naming_conventions.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+# Script used to review jenkins jobs and report job names
+# not following conventions
+# https://github.com/rcbops/rpc-gating#naming-conventions
+
+import argparse
+import os
+import re
+import sys
+import yaml
+
+
+def parse_args():
+    description = "Parses JJB files to ensure naming conventions followed"
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--exclude-files",
+                        required=False,
+                        help="Comma seperated list of files to exclude")
+    parser.add_argument("--dirs",
+                        required=True,
+                        help="Comma seperated dirs to parse")
+    parser.add_argument("--parse-jjb", required=False,
```

Comment:
I hit on this a bit in a reply to @hughsaunders , but my intention was to separate this linting between filenames and JJB's. Rather than have this parse all directories for filenames **and** any job/job-template/project names following standards, to only do either/or per run. I also think this makes it a bit more obvious in `lint.sh` that one run of it is file naming conventions, the other is job naming conventions. 

But, the way that this is called in lint.sh, it could really go either way and I don't have a specific example of when this may include a different directory of JJBs to be linted but not filenames.